### ISSUE TITLE
spark 2.1.0 has issue in parsing CSV with new line character inside a quotes

### ISSUE THREAD

**Issue by dgoutham** [databricks/spark-csv#425]
while loading csv file to dataframe , i observed latest spark version is unable to parse new line characters inside quotes and treating a single column value as two different values and creating extra row.

i have used spark 2.0.2 with spark-csv and it worked.

btw i have univocity parserLib in both cases.

**Comment by HyukjinKwon** [comment]
From 2.2.0, it supports `wholeFile` option for this.

**Comment by aaronsteers** [comment]
@HyukjinKwon - Can you please clarify your comment regarding 2.2? Do you mean this is resolved in a future Spark release? Do you know when this fix might be released, or if there are any workarounds that can be applied in the meanwhile?

Thanks.

**Comment by HyukjinKwon** [comment]
Yes, it will be included from Spark 2.2.0. I don't know the exact release date. My wild guess is about next month.

**Comment by codingnirvana** [comment]
@dgoutham @aaronsteers - I am facing a similar issue after upgrading to 2.1.1. Were you guys able to find workaround for this?

**Comment by aaronsteers** [comment]
@dgoutham - Thanks for your update. This will be very helpful.

@codingnirvana - I was not able to find a solution so I had to implement an upstream removal of all newline characters from the source csv files.

**Comment by aaronsteers** [comment]
@codingnirvana - Per the comment from @dgoutham, it may actually be easier for you to revert to 2.0.2 (if feasible) until the patch is merged into 2.2.

**Comment by codingnirvana** [comment]
@dgoutham @aaronsteers - Thanks. For now, I ended up fixing the upstream system like @aaronsteers. I did think about reverting to 2.0.2 as well, but given that the [PR](https://github.com/apache/spark/pull/16976) to fix this is already part of 2.2.0 RC2 (which was put out for voting last week), I thought its probably better to wait for the new release.

**Comment by stevefuzzy** [comment]
Hmm, I'm still experiencing this issue in June 2018, with v2.3.0.  This was supposedly fixed in 2.2.
What am I missing.

**Comment by praetp** [comment]
Issue still there in v2.3.2.

**Comment by Sleekery** [comment]
In PySpark, multiLine=True doesn't work either. 

>>> newdatadf=spark.read.csv(filepath+filename,multiLine=True,schema=brokenschema,sep='|',header=False,columnNameOfCorruptRecord='broken')
>>> print('Done')
Done
>>> newdatadf.show(1)

And then it never stops working.
